nuget.cake

```diff
@@ -36,14 +36,14 @@ public static void DownloadPackage(this ICakeContext context, DirectoryPath exte
     var packageInfo = (
         from packageItem in package?.items ?? Enumerable.Empty<PackageItem>()
         from item in packageItem.items ?? Enumerable.Empty<PackageRevisions>()
-        where item?.catalogEntry?.version == packageItem.upper
+        where item?.catalogEntry?.listed ?? false
         select new
         {
             id = packageId,
-            version = packageItem?.upper,
+            version = item?.catalogEntry?.version,
             item?.packageContent
         }
-    ).FirstOrDefault();
+    ).LastOrDefault();
```

Comment:
Shouldn't we order the list first to have a guaranteed order in case something changes on nuget API side?

Reply:
Cheers. I would be surprised if the NuGet team changed the sort order of this API as that would break so many tools... But I don't mind being a little conservative either. The only caveat is that we can't simply sort the list by the version string (as `0.10.0` would come before `0.2.0`, for example) so we'll have to use a natural string comparer or a parser that understands version numbers. I think using `NuGetVersion` from [`NuGet.Versioning`](https://www.nuget.org/packages/NuGet.Versioning/) might be the best/safest way to go. I'll update the PR shortly

Reply:
Sorting the list of NuGet packages by NuGetVersion via [`0c87ab3`](https://github.com/cake-build/website/pull/1261/commits/0c87ab3b23aadf6f12fb5c3c66dd6f57f621871c)